tsnet: flake crash in TestPacketFilterFromNetmap

https://github.com/tailscale/tailscale/actions/runs/19255102603/job/55048060953?pr=17831

```
=== RUN   TestPacketFilterFromNetmap
=== PAUSE TestPacketFilterFromNetmap
=== CONT  TestPacketFilterFromNetmap
=== RUN   TestPacketFilterFromNetmap/IP_based_peers
=== PAUSE TestPacketFilterFromNetmap/IP_based_peers
=== RUN   TestPacketFilterFromNetmap/capmap_based_peers
=== PAUSE TestPacketFilterFromNetmap/capmap_based_peers
=== RUN   TestPacketFilterFromNetmap/capmap_based_peers_changed
=== PAUSE TestPacketFilterFromNetmap/capmap_based_peers_changed
=== CONT  TestPacketFilterFromNetmap/IP_based_peers
=== CONT  TestPacketFilterFromNetmap/capmap_based_peers_changed
=== CONT  TestPacketFilterFromNetmap/capmap_based_peers
    tsnet_test.go:117: DERP httpsrv listener: 127.0.0.1:52172
    tsnet_test.go:117: DERP httpsrv listener: 127.0.0.1:52170
    tsnet_test.go:130: testcontrol listening on http://127.0.0.1:52176/
    tsnet_test.go:117: DERP httpsrv listener: 127.0.0.1:52171
    tsnet_test.go:130: testcontrol listening on http://127.0.0.1:52178/
    tsnet_test.go:130: testcontrol listening on http://127.0.0.1:52177/
2025/11/11 04:40:48 tsnet starting with hostname "node", varRoot "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestPacketFilterFromNetmapcapmap_based_peers4017940094\\001\\node"
2025/11/11 04:40:48 tsnet starting with hostname "node", varRoot "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestPacketFilterFromNetmapIP_based_peers1510784689\\001\\node"
2025/11/11 04:40:48 tsnet starting with hostname "node", varRoot "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\TestPacketFilterFromNetmapcapmap_based_peers_changed2119568512\\001\\node"
2025/11/11 04:40:48 LocalBackend state is NeedsLogin; running StartLoginInteractive...
2025/11/11 04:40:48 LocalBackend state is NeedsLogin; running StartLoginInteractive...
2025/11/11 04:40:48 LocalBackend state is NeedsLogin; running StartLoginInteractive...
    testcontrol.go:315: Created user userid:1 for node nodekey:53e5120545f48533a1d80129e47ed6265614f97d4a4e055001115736993bef7a
    testcontrol.go:315: Created user userid:1 for node nodekey:c7d235278463555f71987b1addffbfa1ff1f2793f8568962b5a9cf2522680221
    testcontrol.go:315: Created user userid:1 for node nodekey:ab72c8806b862ecffd835f08afba9488e7ad586a5b0223fc3cbd1c7b191a7a4a
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x72 pc=0x7ff69448dbcc]

goroutine 7012 [running]:
net/http.(*persistConn).writeLoop(0xc001191440)
	C:/hostedtoolcache/windows/go/1.25.3/x64/src/net/http/transport.go:2624 +0x291
created by net/http.(*Transport).dialConn in goroutine 7010
	C:/hostedtoolcache/windows/go/1.25.3/x64/src/net/http/transport.go:1948 +0x17a5
FAIL	tailscale.com/tsnet	9.018s
```

--- a/ipn/desktop/sessions_windows.go
+++ b/ipn/desktop/sessions_windows.go
@@ -510,10 +510,13 @@ func sessionWatcherWndProc(hWnd windows.HWND, msg uint32, wParam, lParam uintptr
 }
 
 func pumpThreadMessages() {
-	var msg _MSG
-	for getMessage(&msg, 0, 0, 0) != 0 {
-		translateMessage(&msg)
-		dispatchMessage(&msg)
+	var p runtime.Pinner
+	defer p.Unpin()
+	msg := &_MSG{}
+	p.Pin(msg)
+	for getMessage(msg, 0, 0, 0) != 0 {
+		translateMessage(msg)
+		dispatchMessage(msg)
 	}
 }
 

